server/handlers/order.js

```diff
@@ -56,7 +56,29 @@ exports.createOrderHandler = async function (req, res, next) {
     console.error(`CreateOrder() -> couldn't retrieve newly created order`);
     return res.status(404).end();
   }
+  //GET CLIENT WALLET
+  let client;
 
+  try {
+    client = await dao.getClientByID(req.body.clientID.toString());
+  } catch (err) {
+    return res.status(500).end();
+  }
+  if (!client) {
+    console.error(
+      `getClientByID() -> couldn't retrieve client`
+    );
+  }
+
+  //IF WALLET NOT ENOUGH SET THE STATE TO NOT COVERED
+  if(client.wallet < totalPrice){
+    try{
+      result = await dao.updateOrderStatus(result._id, "not covered");
+      result = result.value;
+    }catch(err){
+      return res.status(500).end();
+    }
+  }
```

Comment:
Instead of modify the order after it was inserted in the DB is better to modify it before it is written, in this way we don't have to access the database multiple time